### PR TITLE
Add sdist check to `spin sdist`

### LINKED ISSUE #7437: `tools/check_sdist.py` is currently unused
Either remove it completely or integrate it into our CI again if it still works.

### PR DESCRIPTION
**Pull request by stefanv**
Closes #7437

### CODE REVIEW

File: .spin/cmds.py [hunk 1]
```diff
@@ -99,3 +99,18 @@ def ipython(ctx, ipython_args):
...
+@click.command()
+def sdist():
+    """📦 Build a source distribution in `dist/`"""
+    p = util.run(["pyproject-build", ".", "--sdist"], output=False)
+    built_line = next(
+        line
+        for line in p.stdout.decode('utf-8').split('\n')
+        if line.startswith('Successfully built')
+    )
+    print(built_line)
```
Comment (lagru) [inline .spin/cmds.py:113]:
```suggestion
    output = p.stdout.decode("utf-8")
    if "Successfully built" in output:
        print(built_line)
    else:
        print(output)
```
Seems a bit clearer.

**Review by lagru — APPROVED** [review]
Two small suggestions, LGTM otherwise. :)

File: .spin/cmds.py [hunk 2]
```diff
@@ -99,3 +99,18 @@ def ipython(ctx, ipython_args):
     ) + ipython_args
 
     ctx.forward(meson.ipython)
+
+
+@click.command()
+def sdist():
+    """📦 Build a source distribution in `dist/`"""
+    p = util.run(["pyproject-build", ".", "--sdist"], output=False)
```
[the remark this answers is not in the document]
Comment (stefanv) [reply .spin/cmds.py:107]: Unfortunately, cannot capture and print the output simultaneously without quite a bit more effort.

File: .spin/cmds.py (hunk 1 above)
Comment (stefanv) [reply .spin/cmds.py:113]: The run command prints output upon failure.

**Review by jarrodmillman — APPROVED** [review]

**Comment by stefanv** [comment]
I think GitHub confused this PR with the test branch I pushed up as `maintenance/test-sdist-via-spin` :thinking: 

Either way, looks like our wheel builder may be giving us issues on Windows, so better sort that out for the release.

**Review by lagru — APPROVED** [review]

**Pull request merged**